cosmrs/src/base.rs

```diff
@@ -72,7 +72,7 @@ impl FromStr for AccountId {
     fn from_str(s: &str) -> Result<Self> {
         let (hrp, bytes) = bech32::decode(s).wrap_err("failed to decode bech32")?;
 
-        if bytes.len() == tendermint::account::LENGTH {
+        if bytes.len() >= tendermint::account::LENGTH {
```

Comment:
Seems like this should perhaps check inclusion within a set of allowed lengths?

```suggestion
        if matches!(bytes.len(), tendermint::account::LENGTH | 32) {
```

(seems like perhaps there should be a new upstream constant for 32-byte lengths, but we can cross that bridge when we get there)

Reply:
Well, I guess the decision is ultimately up to you. I was inspired by the associated error message, i.e. `account ID should be at least {} bytes long, but was {} bytes long` implying that the check should be performed against a range rather than against a set of allowed values. Let me know which one you prefer and I'll update the PR accordingly!

Reply:
Furthermore, I just looked at `cosmos-sdk` code directly and it looks like for their `AccAddress` (which I think the `AccountId` corresponds to?) they only length check they perform is whether the bytes decoded from the bech32 representation are shorter than 255 (and obviously whether they're not empty) (https://github.com/cosmos/cosmos-sdk/blob/f133317cfe8b4723ccb5e3687ffc9ab71b52ea75/types/address.go#L148-L150)

Reply:
I've updated the PR accordingly. Presumably `new` and `to_bytes` would also need to be changed, however, those would be incompatible breaking changes and thus I didn't want to touch them now : )